doc/source/developing/releasing.rst

```diff
@@ -76,92 +122,88 @@ In short, a manual backport consist of 4 steps
+
+For a major release that includes removal of deprecated features, make sure that all
 deprecated features targeted for removal in the new release are removed from the
-``main`` branch, ideally in a single PR. Such a PR can be issued at any point
-between the previous minor or major release and the new one. Then, all that
-needs to happen is the ``main`` branch must get merged into the ``stable``
-branch, and any conflicts that happen must be resolved, almost always in favor
-of the state of the code on the ``main`` branch.
+``main`` branch prior to creating the new series branch (:ref:`doing-a-minor-or-major-release`).
+Ideally, the deprecated features should be removed in a single PR.
 
 
-Incrementing Version Numbers and Tagging a Release
-~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+Incrementing Version Numbers
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~
 
 Before creating the tag for the release, you must increment the version numbers
 that are hard-coded in a few files in the yt source so that version metadata
-for the code is generated correctly. This includes things like ``yt.__version__``
-and the version that gets read by the Python Package Index (PyPI) infrastructure.
-
-The paths relative to the root of the repository for the three files that need
-to be edited are:
+for the code is generated correctly. The paths relative to the root of the
+repository for the three files that need to be edited are:
 
 * ``doc/source/conf.py``
 
   The ``version`` and ``release`` variables need to be updated.
 
-* ``setup.py``
+* ``pyproject.toml``
 
-  The ``VERSION`` variable needs to be updated
+  The ``version`` variable needs to be updated
 
-* ``yt/__init__.py``
+* ``yt/_version.py``
 
   The ``__version__`` variable must be updated.
 
-Once these files have been updated, commit these updates. This is the commit we
-will tag for the release.
+To update these files, check out and update the branch that will be released (``main``
+if doing a major or minor release, or the backport branch if doing a bugfix release).
 
-To actually create the tag, issue the following command from the ``stable``
-branch:
+Once these files have been updated, commit these updates and submit a pull request
+targeting the branch that will be released. This is the commit we
+will tag for the release. After a major or minor release is complete, the version numbers
+on ``main`` should be updated to reflect development version via another pull request.
+
+
+.. _tagging-a-release:
+
+Tagging a Release
+~~~~~~~~~~~~~~~~~
+
+After incrementing version numbers, checkout and update the branch that will be released
+and actually create the tag by issuing the following command:
 
 .. code-block:: bash
 
    git tag <tag-name>
 
 Where ``<tag-name>`` follows the project's naming scheme for tags
 (e.g. ``yt-3.2.1``). Once you are done, you will need to push the
-tag to github::
+tag to the yt repository on github::
 
-  git push origin --tag
+  git push upstream --tag
```

Comment:
note that `gh release create` lets you tag the commit interactively. I'm not 100% sure how it integrates with draft notes but I'd assume the experience to be consistent, so maybe you want to revisit this to use `gh` as much as possible ?